task_processing/plugins/mesos/execution_framework.py

```diff
@@ -609,7 +609,19 @@ def resourceOffers(self, driver, offers):
 
                         )
                     )
+                    # Emit the staging event for successful launches
                     if not task_launch_failed:
+                        update = Dict(
```

Comment:
i know why we need this, but it feels too arbitrary. we could instead attach task offer to each status update (or at least to task_running) in the framework - this way status updates are all uniform and every executor has the same data to work with.

Reply:
Will fix.